src/System.Runtime/tests/System/EnumTests.cs

```diff
@@ -11,24 +11,29 @@ namespace System.Tests
 {
     public static partial class EnumTests
     {
+        public static IEnumerable<object[]> Parse_TestData()
+        {
+            yield return new object[] { "Red", false, SimpleEnum.Red };
+            yield return new object[] { " Red", false, SimpleEnum.Red };
+            yield return new object[] { "Red ", false, SimpleEnum.Red };
+            yield return new object[] { " red ", true, SimpleEnum.Red };
+            yield return new object[] { "B", false, SimpleEnum.B };
+            yield return new object[] { "B,B", false, SimpleEnum.B };
+            yield return new object[] { " Red , Blue ", false, SimpleEnum.Red | SimpleEnum.Blue };
+            yield return new object[] { "Blue,Red,Green", false, SimpleEnum.Red | SimpleEnum.Blue | SimpleEnum.Green };
+            yield return new object[] { "Blue,Red,Red,Red,Green", false, SimpleEnum.Red | SimpleEnum.Blue | SimpleEnum.Green };
+            yield return new object[] { "Red,Blue,   Green", false, SimpleEnum.Red | SimpleEnum.Blue | SimpleEnum.Green };
+            yield return new object[] { "1", false, SimpleEnum.Red };
+            yield return new object[] { " 1 ", false, SimpleEnum.Red };
+            yield return new object[] { "2", false, SimpleEnum.Blue };
+            yield return new object[] { "99", false, (SimpleEnum)99 };
+            yield return new object[] { "-42", false, (SimpleEnum)(-42) };
+            yield return new object[] { "   -42", false, (SimpleEnum)(-42) };
+            yield return new object[] { "   -42 ", false, (SimpleEnum)(-42) };
```

Comment:
Is there a reason to prefer this approach over the InlineData approach? I don't have a strong preference I'm just trying to understand the motivation